observations/obs_converters/ARVOR/arvor_to_obs.f90

```diff
@@ -0,0 +1,390 @@
+real(r8)           :: obs_error_sal     = 0.02_r8  ! ARVOR CTDs are precise
+logical            :: debug             = .true.
+
+
+namelist /arvor_to_obs_nml/ file_in,          &
+                            file_list,        &
+                            obs_error_temp,   &
+                            obs_error_sal,    &
+                            file_out,         &
+                            avg_obs_per_file, &
+                            debug
+
+! Start Converter
+call initialize_utilities()
+
+! Read the namelist options
+call find_namelist_in_file('input.nml', 'arvor_to_obs_nml', iunit)
+read(iunit, nml = arvor_to_obs_nml, iostat = io)
+
+if (do_nml_file()) write(nmlfileunit, nml=arvor_to_obs_nml)
+if (do_nml_term()) write(     *     , nml=arvor_to_obs_nml)
+
+! Set the calendar kind
+call set_calendar_type(GREGORIAN)
+
+! Check the files
+if (file_in   /= '' .and. file_list /= '') then
+   string1 = 'One of input ARVOR file or the file list must be NULL'
+   call error_handler(E_ERR, source, string1)
+endif
+if (file_list /= '') from_list = .true.
+
+! Get number of observations
+num_new_obs = avg_obs_per_file
+if (from_list) then
+   call find_textfile_dims(file_list, nfiles)
+   num_new_obs = avg_obs_per_file * nfiles
+endif
+
+! Assign obs errors
+float_errs = [obs_error_temp, obs_error_sal]
+
+! Initialize obs seq file
+call static_init_obs_sequence()
+call init_obs(obs, num_copies, num_qc)
+call init_obs(prev_obs, num_copies, num_qc)
+
+print *
+if (file_exist(file_out)) then
+   write(*, '(A)') 'Output file: '//trim(adjustl(file_out))//' exists. Replacing it ...'
+else
+   write(*, '(A)') 'Creating "'//trim(adjustl(file_out))//'" file.'
+endif
+
+call init_obs_sequence(obs_seq, num_copies, num_qc, num_new_obs)
+call set_copy_meta_data(obs_seq, num_copies, 'ARVOR observation')
+call set_qc_meta_data(obs_seq, num_qc, 'ARVOR QC')
+
+! Loop over the obs files
+filenum = 1
+obs_qc  = 0.0_r8
+
+FILELOOP: do
+  ! Get the data file
+  if (from_list) then
+     next_infile = get_next_filename(file_list, filenum)
+  else
+     next_infile = file_in
+     if (filenum > 1) next_infile = ''
+  endif
+  if (next_infile == '') exit FILELOOP
+  print *
+  if (debug) write(*, '(A, i0, X, A)') 'Input file: #', filenum, trim(next_infile)
+
+  ! Read data from file
+  call get_profile_data(next_infile)  
+
+  ! Add obs to sequence
+  call fill_obs(obs, prev_obs, prev_time, obs_qc)
+
+  call cleanup
+
+  filenum = filenum + 1
+enddo FILELOOP
+
+call finish_obs()
+call finalize_utilities()
+
+
+contains
+
+
+!-----------------------------------------------
+! Read the data profile and the associated dates 
+subroutine get_profile_data(filename)
+
+character(len=*), intent(in) :: filename
+character(len=*), parameter  :: routine = 'get_profile_data'
+
+integer :: i, k, nobs
+
+! Open csv file and get dims
+call csv_open(filename, cf, routine)
+nobs = cf%nrows
```

Comment:
it's better to have an accessor function that returns the number of rows rather than accessing the cf structure directly.  then the handle contents can be private and calling code can't accidentally change them.